### PR TITLE
267: WHOIS search page

### LINKED ISSUE #267: Implement WHOIS / registry info search page on get.gov
### Story

A WHOIS search feature needs to be added on a new page in get.gov/domains/whois.

[Updated Figma](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=5943-73158)

### Acceptance Criteria

- [x] [A new page for WHOIS search](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?node-id=3743%3A23903&mode=dev) is added under /domains 
- [x] When the user [searches a domain that is registered, they are shown WHOIS data](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?node-id=3991%3A4345&mode=dev). 
- [x] Certain data points are pulled from the raw data and styled according to the mockup. Registrar info (Name, Security email, Whois server, Status; Dates (Created, Expires, Updated), Name servers (list name servers + IPs inside of parentheses)
- [x] User can only search .gov domains. If they search another TLD, like example.com, [they are shown the error: "You can only search in domain names that end in .gov"](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?node-id=3830%3A55128&mode=dev)
- [x] If user searches domain that isn't registered, they see the blue info alert: ["<domain.gov> isn't a registered domain"](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?node-id=3984%3A1558&mode=dev)
- [x] Below search field, there's a line: "Want to see if a domain is available? [Check availability here.](https://get.gov/domains/choosing/)

- [x] On get.gov/domains/choosing, add a link to the unavailable error message 
 as shown in [this figma](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=7325-18746&t=B3yGu5OSzHbZD7nC-0) that links to the whois page 

### Additional Context

The [google doc for content](https://docs.google.com/document/d/1CwrBMTXnxdde3Il4P0dFAkWFvHyqv2qdSyfvg0NsG4k/edit?usp=sharing) is the source of truth for the content for this page

For devs:
You could use a package from online to help with whois as there are a few. However, if you don't find one that's reliable, you could send the whois message and either parse into a json or do string search to find the values that need to display at the top. Then show the full string output for the full whois section. While string search is not great, for a few values it may be reasonable to do if json seems to difficult. 

While not many of these node packages for whois are well maintained, you could use their code as examples for writing your own as most are a single file for executing this function. 

Note on difficulty, please keep prod team updated if blockers or issues arise that slow this down
Below is a 5 mins js function to retreive whois, this is just an **example** of how to get whois and retrieve the data as string without any third party dependencies. note net comes with node.

You may copy the below code into a .js, such as whois.js file then run "node whois.js" in terminal. 

 
```
const net = require('net');
const domain = 'gsa.gov';

// Define the Cloudflare's WHOIS server
const whoisServer = 'whois.dotgov.gov';
function parseWhois(whoisString) {
    // super quick non elegant parsing 
    // Split the string into lines
    const lines = whoisString.split('\n');
    const result = {};
  
    // Iterate over each line
    lines.forEach(line => {
      // Check if the line contains a colon
      const colonIndex = line.indexOf(':');
      if (colonIndex !== -1) {
        // Extract the key and value, trim whitespace
        const key = line.substring(0, colonIndex).trim();
        const value = line.substring(colonIndex + 1).trim();
        // Add the key-value pair to the result object
        //missing here you need to handle multiple nameservers
        result[key] = value;

        //note this is quick and has bad parsing on << >> values
        //I also didn't convert to lower case, which should be done. 
        //This should just be serve as a quick an example
      }
    });
  
    return result;
  }
// Function to send WHOIS query
function sendWhoisQuery(domain, server, callback) {
    const client = net.createConnection({ port: 43, host: server }, () => {
        client.write(`${domain}\r\n`);
    });

    let whoisData = '';
    let whoisJson={}
    client.on('data', (data) => {
        whoisData += data.toString();
        //convert to json and just print here
        // in the real code you will want to return data as json so that you can access the values
        whoisJson =parseWhois(whoisData)
        console.log(JSON.stringify(whoisJson))
    });


    client.on('end', () => {
        callback(null, whoisData);
    });

    client.on('error', (err) => {
        callback(err);
    });
}

// Perform the WHOIS query
sendWhoisQuery(domain, whoisServer, (err, data) => {
    if (err) {
        console.error('Error:', err);
    } else {
       // totally complex way to just prove the whois shows as a string
        console.log('WHOIS data:', data);
    }
});

```
### Issue Links

Design ticket: cisagov/manage.get.gov#1221

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #267 and #316 

## Changes

- Completes #316 - Use manage.get.gov RDAP endpoint to request WHOIS data
- Add WHOIS page to get.gov
- Add WHOIS search and search results to WHOIS page. Search results display RDAP data of a .gov domain.
- Handle user error messages when a searched domain is not a .gov domain (either by 1. domain not ending in .gov or 2. .gov domain not existing).
- When a user searches an unavailable .gov domain on the availability search, the user is given the option to search the domain's WHOIS data on the WHOIS page.

## Context for reviewers

Enabling URLs of specific domain queries on the WHOIS page will be addressed in #313 which handles supporting query variables on the 11ty site.

Given that we use RDAP to pull domain data instead of WHOIS but WHOIS is the more recognized protocol/term, there are considerations for the language that we use to communicate this information to the user.

Cloudflare RDAP currently does not include IP addresses of nameservers. Example RDAP nameserver data below:
```
{
      "objectClassName": "nameserver",
      "handle": "DF12BBE1F-ROID",
      "ldhName": "ernest.ns.cloudflare.com",
      "links": [
        {
          "href": "https://rdap.cloudflareregistry.com/rdap/nameserver/ernest.ns.cloudflare.com",
          "rel": "self",
          "type": "application/rdap+json"
        }
      ],
      "remarks": [
        {
          "description": [
            "Summary data only. For complete data, send a specific query for the object."
          ],
          "title": "Incomplete Data",
          "type": "object truncated due to unexplainable reasons"
        }
      ]
    }
```

## Setup

Go to the [WHOIS page on cloud pages ](https://federalist-877ab29f-16f6-4f12-961c-96cf064cf070.sites.pages.cloud.gov/preview/cisagov/get.gov/es/267-whois-page/domains/whois/) and test the following:

- [ ] User can search WHOIS/RDAP data for an existing .gov domain. Search results display data for a domain's contact information, statuses, events, nameservers, and notices.
- [ ] User is given appropriate error messages for invalid domain search inputs. Queries for domains not ending in .gov inform the user their searched domain must end in .gov. Queries for .gov domains that do not exist inform the user that their searched domain is not a .gov website.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [x] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

### CODE REVIEW

File: _includes/meta.html
Comment (erinysong) [reply _includes/meta.html:171]: Does anyone have suggestions on how to handle this warning? This is for existing code and will be looking while this is in review, but opening this thread for any suggestions

**Comment by kristinacyin** [comment]
Do you know where I can find the reasoning for the content? In particular, I feel like the following section is overly technical for the average visitor:
> This is a web version of our WHOIS, which you can also access in a command-line interface application like Terminal ($ whois -g get.gov)

Is it better to move this page to where it can be more easily accessed? Seems to be a bit buried in the Domains section of the website, it might make sense to have it in the navigation bar (maybe in between "Domains" and "Blog" instead?

**Comment by h-m-f-t** [comment]
A few thoughts:
* To @kristinacyin's 1 and 2, I do think we should keep the command-line instruction (let's educate users!) but perhaps make it `monospaced `. I also don't think we should add this to the top-nav.
* We should still call this "WHOIS" but in the blurb we should say we're actually using RDAP. I can help revise.
* Stuff I think we should remove:
  * The `version` value from all elements. 
  * `Reregistration` from Events
  * `Handle`s from Nameservers (I think we typically use "name servers")
  * Notices. I could see us being valuable in a command-line RDAP tool, but not here.
* It's not easy to see when multiple successful queries have been returned. For names that exist, what if we added it as a header?
* The .gov at the end makes me think I could search for a string without a TLD suffix, but it throws an error. This is different than how we treat queries in "choosing your .gov domain name"
image

* On the sidenav, let's change "WHOIS domain lookup" to just "WHOIS lookup".
* In the error text, let's say "You can learn about who's registered this domain by doing a WHOIS lookup" instead.
  * ...We don't actually show who owns domains in WHOIS, since we redact those values. I could support us un-redacting the fields we publish in current-full.csv (as a different issue).

image

**Comment by kristinacyin** [comment]
> There's a lot of feedback here on changing design/content. Does this need deeper eval than just a typical design review?

I think I'm just missing some context here. I know @gabydisarli and Erin already discussed the design extensively, so I'm not asking anything to be changed. I'm more curious about the reasoning since that might mean we should change the "Choosing your .gov domain name" page to match this design.

**Comment by erinysong** [comment]
thank you for the thoughtful comments @kristinacyin and @h-m-f-t. For content, we only discussed switching from WHOIS to RDAP to pulling domain data so I'm open to any suggestions for content changes

**Comment by erinysong** [comment]
> * Is it better to move this page to where it can be more easily accessed? Seems to be a bit buried in the Domains section of the website, it might make sense to have it in the navigation bar (maybe in between "Domains" and "Blog" instead?
> *

re: your questions in order
1. I also would lean towards updating that content line, especially since that command is WHOIS and would output slightly differently formatted data that RDAP
2. I may not have enough context on how often users will want to use the WHOIS page to have an opinion on if this an accessible enough location on get.gov to use this - open to whatever suggestions we have if there are any decision changes on this!
3. @gabydisarli and I discussed how the .gov search was different on Choosing domains (I believe that search bar was added at a time we had different designs for the .gov search bar). I think this would be best to update in another PR for the sake of keeping items separate - if we want to change the Choosing domains search bar and there's no ticket yet, I'm happy to make one!

thank you for the thorough questions - super appreciated and agree with the points that you made!

**Comment by kristinacyin** [comment]
> re: your questions in order
> 
> 1. I also would be open to updating that content line, especially since that command is WHOIS and would output slightly differently formatted data that RDAP. I can look into if there are similar commands for RDAP but I think there's an underlying issue of WHOIS vs. RDAP in the content we use on this page.
> 2. I may not have enough context on how often users will want to use the WHOIS page to have an opinion on if this an accessible enough location on get.gov to use this - open to whatever suggestions we have if there are any decision changes on this!
> 3. @gabydisarli and I discussed how the .gov search was different on Choosing domains (I believe that search bar was added at a time we had different designs for the .gov search bar). I think this would be best to update in another PR for the sake of keeping items separate - if we want to change the Choosing domains search bar and there's no ticket yet, I'm happy to make one!
> 
> thank you for the thorough questions - super appreciated and agree with the points that you made!

@erinysong No, thank you for helping to clear up my confusion!

1. Yes, it seems like the content in general needs to be updated based on @h-m-f-t's points. I'm assuming you and Cameron will work together to change WHOIS to RDAP. 
2. I think we can keep it where it is for now. If we get feedback that users can't find it / a lot users use it, then we can consider moving it then. 
3. Let me check in with Gaby. If needed, I can create the ticket! 😄

File: _includes/domain-search.html
Comment (erinysong) [inline _includes/domain-search.html:23]: Renamed class name given this variant (light blue) will only be when a domain search bar is on a blue background which will be less common than our new gray variant for white backgrounds

File: _includes/meta.html
Comment (erinysong) [inline _includes/meta.html:171]: Fixed HTML render helper when we aren't appending new content. I didn't see examples where we're clearing the inner HTML of an element to an empty div but if the need arises, I suggest making a new method to clear the display of an HTML and just making the inner html to an empty string rather than adding an empty div

File: _includes/whois-search.html
Comment (Matt-Spence) [inline _includes/whois-search.html:167]: Append is false by default isn't it?

File: _includes/whois-search.html
Comment (Matt-Spence) [inline _includes/whois-search.html:180]: Same thing

File: _includes/meta.html
Comment (Matt-Spence) [reply _includes/meta.html:171]: I'm gonna pause my review and look into this deeper.

File: package.json
Comment (erinysong) [inline package.json:53]: seems to be an addition to USWDS - will investigate since I'm not too sure how it got here

File: _includes/domain-availability/domain-availability-search-input.html
Comment (Matt-Spence) [inline _includes/domain-availability/domain-availability-search-input.html:1]: hat's the reasoning behind putting this and the availability button in the includes?

**Review by Matt-Spence — COMMENTED** [review]
Pausing while I look deeper into the DOM text issue.

File: styles/_uswds-theme-custom-styles.scss
Comment (erinysong) [inline styles/_uswds-theme-custom-styles.scss:341]: found @extend useful in creating CSS "variants" if anyone wants to make variants of elements in the future

**Comment by AnnaGingle** [comment]
Quick accessibility note 
The input's ANDI or screenreader output is `Search domain WHOIS input`. This fails section 508 success criteria 1.3.1-programmatic-label. The screen reader / ANDI output should be: `.gov domain WHOIS lookup`
Screenshot 2024-08-13 at 2 51 44 PM

File: _includes/domain-availability-highlight.html
Comment (erinysong) [reply _includes/domain-availability-highlight.html:2]: good catch thank you! copied this comment from referencing another template and good chance to fix it everywhere

File: _includes/domain-availability/domain-availability-search-input.html
Comment (erinysong) [reply _includes/domain-availability/domain-availability-search-input.html:1]: Rechecked and `check-domain-availability-button.html` is an artifact of a previous conversation when discussing possible options to isolate our search templates that resulted in us deciding that should be in the scope of a separate issue. Since it's not referenced anywhere, will delete `check-domain-availability-button.html` for this PR - thank you for catching that remaining artifact!

File: _includes/whois-search.html
Comment (erinysong) [reply _includes/whois-search.html:167]: It's true by default but I also feel like from the way we may use it in the future, it may make more sense to make it false by default. @zandercymatics I believe you may have made this method (correct me if I'm wrong) so was curious if there's any reason we'd want to default append to one or the other?

**Comment by erinysong** [comment]
> Reference i

Thank you for catching and posting the accessibility issue @AnnaGingle! Just corrected the aria-text and title of the search input. For future cases, is there a good source of truth for what aria labels should be given to new UI additions?

File: styles/_uswds-theme-custom-styles.scss
Comment (erinysong) [inline styles/_uswds-theme-custom-styles.scss:419]: Previously existing workaround for misalignment on USWDS icons on their alerts -- applied this to be global since we're adding more USWDS alerts in this PR

**Comment by erinysong** [comment]
> @erinysong I'll be doing your design review

thank you @gabydisarli ! If there are any updated Figma links I should reference, feel free to let me know since I know this issue originally was assigned before we reorganized our Figma :")

File: _includes/domain-availability/domain-availability-search-input.html
Comment (Matt-Spence) [inline _includes/domain-availability/domain-availability-search-input.html:19]: The search box is a bit confusing. It looks like the .gov is being supplied by the search box, but actually it's required. Maybe this is a question for design, but should we auto-append the required .gov at the end?

File: _includes/whois-search.html
Comment (Matt-Spence) [inline _includes/whois-search.html:37]: Are there ever going to be instances where this information isn't redacted? Every single test I tried had all this redacted. That's not really a problem, I'm just wondering.

**Review by Matt-Spence — COMMENTED** [review]
The functionality of this is great, but there are some concerns that need to be addressed before this gets merged. It may seem like there's a lot of them, but basically they're all different permutations of the same thing: we shouldn't treat RDAP data as trustworthy. That data needs to be either sanitized or injected as text rather than rendered as html.

File: _includes/meta.html
Comment (erinysong) [inline _includes/meta.html:22]: Does anyone know if there's a way to supply a CDNJS string with the latest version of a package? I'm not a huge fan of how the version needs to be inputted into the src URL to access DOMPurify

Comment (h-m-f-t) [reply _includes/meta.html:22]: Is there a reason we're using a CDN for this instead of self-hosting?

File: _includes/whois-search.html
Comment (erinysong) [reply _includes/whois-search.html:37]: I think it's likely most of the data here will be redacted due to the nature of PII - I was also curious how helpful this data would be since I also found the primarily non-redacted data were the different dates of registration/expiry

Comment (erinysong) [reply _includes/whois-search.html:37]: I think it's likely most of the contact data here will be redacted due to the nature of PII - I was also curious how helpful this data would be since I also found the primarily non-redacted data were the different dates of registration/expiry

Comment (erinysong) [reply _includes/whois-search.html:37]: @h-m-f-t do you happen to know if it's likely if this WHOIS contact information will not be redacted? I can do some digging to see if this is redacted by default on Cloudflare's end or if it's getting redacted for another reason

**Comment by erinysong** [comment]
Design feedback to-dos

Styling
- [ ] Add "WHOIS Results for [domain_name]" h2 above WHOIS cards. 
- [ ] Adjust heading hierarchies to adjust added above heading. Change card titles from h2 to h3. Change subtitles from h3 to h4 (18px) and keep blue color.
- [ ] Change card bottom padding to 16
- [ ] Change usa-card__header top padding to 16px
- [ ] Remove margins from card subheadings (Registrar, administration, registrant)

Validation

- [ ] Allow users to WHOIS search WHOIS names without appending .gov
- [ ] Remove version data from Contact Information
- [ ] Verify if nameserver handle is useful

File: _includes/whois-search.html
Comment (h-m-f-t) [reply _includes/whois-search.html:40]: We do redact this data by default, except for `Security Email:`. While it's possible we will display some of this information in the future (most likely some of the `Registrant...` info, we don't currently have plans to do so.

File: _includes/meta.html
Comment (h-m-f-t) [reply _includes/meta.html]: Using a CDN for this during development makes total sense, but we should self-host this if we elect to use it.

Comment (h-m-f-t) [reply _includes/meta.html]: @erinysong, leads chatted in refinement, and we're good to see this land in this PR, but can you create an issue to in-source this dependency before this PR is merged? See https://github.com/cisagov/manage.get.gov/issues/2826

Comment (erinysong) [reply _includes/meta.html]: I think since GIthub's suggestion to sanitize our HTML was called on pre-existing code, this can go in another ticket. I'll uninstall DOMPurify for now and move this to #335 - currently our domain validation prevents users from inputting script characters that can be used in most XSS attacks but we should definitely see if there's a way we can host DOMPurify ourselves

Comment (erinysong) [reply _includes/meta.html]: @h-m-f-t sorry didn't see your most recent comment until I refreshed  - will take a look at 2826 and update #335 as needed

**Comment by gabydisarli** [comment]
This is looking so much better Erin, thank you! A few small tweaks. Can you take a look at the [Figma board](https://www.figma.com/design/qeWM03sfjXgHBHB23rsD6X/get.gov?node-id=9066-134467&node-type=frame&t=LOnd1yJlhBpBNCqG-11) for the changes?

**Comment by h-m-f-t** [comment]
> - [ ] Verify if nameserver handle is useful

It is not! I'd love to omit that.

**Comment by erinysong** [comment]
> > * [ ]  Verify if nameserver handle is useful
> 
> It is not! I'd love to omit that.

Will do! @gabydisarli do we want to edit how the nameservers display if the nameservers card only displays their names now?

**Comment by gabydisarli** [comment]
> > > * [ ]   Verify if nameserver handle is useful
> > 
> > 
> > It is not! I'd love to omit that.
> 
> Will do! @gabydisarli do we want to edit how the nameservers display if the nameservers card only displays their names now?

Yes, please remove that. I adjusted the Figma to remove handle as well.

I also added in the Notices section because I'd like to remove the indenting on the body text there. Can we also hyperlink the links presented there?

**Comment by gabydisarli** [comment]
Can we also move the `Want to see if a domain is available? Check availability here. `to just below the field and before the search results?

**Comment by erinysong** [comment]
thank you @gabydisarli ! Will wait to confirm we still want notices before adding in links.
Will work on a fix for moving `Want to see if a domain is available?...` text below the field and search results. Will need to refactor a bit since the field and search results exist under the same HTML but I can circle back if that takes longer than expected

**Comment by erinysong** [comment]
@h-m-f-t wanted to ask if we ended up deciding to not include the notices section - I believe it's the same RDAP terms of service that get appended to each API response

**Comment by h-m-f-t** [comment]
I'd love to disappear the notices section! 🗑️ @erinysong

**Comment by h-m-f-t** [comment]
@erinysong: Note that there's a TODO on /domains/. Do you have language for that, or do you need some? (cc: @gabydisarli)
image

**Comment by erinysong** [comment]
Notices removed, thanks for the ok and good catch on the excerpt! @gabydisarli that hopefully removes the need to linkify the links in the notices as a side effect :"). If there's any copy I can access for the excerpt, that would be great - I currently have [this doc](https://docs.google.com/document/d/1CwrBMTXnxdde3Il4P0dFAkWFvHyqv2qdSyfvg0NsG4k/edit?tab=t.0) as reference and can make changes as needed

**Comment by erinysong** [comment]
Separately, I noticed that RDAP includes links to their nameservers when listing a domain's nameservers. Is this data something that'd be useful in the future? It seems that the nameserver API leads to identical info as the domain but wanted to call it out in case it'd be useful
RDAP lookup below. Link to [nameserver](image) link
image

**Comment by h-m-f-t** [comment]
Another suggestion: Can this just say "Results for $domain" and drop the "WHOIS"?
image

**Comment by h-m-f-t** [comment]
> If there's any copy I can access for the excerpt, that would be great

Look up information about registered domains (some fields are redacted).

Maybe?

**Comment by erinysong** [comment]
@gabydisarli Would you be open to moving condensing the redacted for privacy values to another ticket? I can create one if you're all right with us handling that separately!

**Comment by gabydisarli** [comment]
> @gabydisarli Would you be open to moving condensing the redacted for privacy values to another ticket? I can create one if you're all right with us handling that separately!

Of course!

**Comment by gabydisarli** [comment]
> Notices removed, thanks for the ok and good catch on the excerpt! @gabydisarli that hopefully removes the need to linkify the links in the notices as a side effect :"). If there's any copy I can access for the excerpt, that would be great - I currently have [this doc](https://docs.google.com/document/d/1CwrBMTXnxdde3Il4P0dFAkWFvHyqv2qdSyfvg0NsG4k/edit?tab=t.0) as reference and can make changes as needed

Thanks Erin, this is the copy for the domains page we had approved a while ago. 

> 
> WHOIS domain lookup
> Search for a .gov domain to get information about who manages it and other registry details.

`[Content doc](https://docs.google.com/document/d/1JjrJS2eYPGWjBCsOIEzITI57kE1iyekiXTAzlQ_efWU/edit?tab=t.0)`

**Review by gabydisarli — APPROVED** [review]
design approved after these remaining changes are made, thanks Erin!

**Comment by erinysong** [comment]
Awesome thank you @gabydisarli ! The page looks so much better. Made a ticket to shorten the redacted contact info here: #337

File: _includes/domain-availability/domain-availability-search-input.html
Comment (erinysong) [reply _includes/domain-availability/domain-availability-search-input.html:19]: Update: Talked with design and the search supports inputs with or without appending .gov as the TLD
image
image

**Review by Matt-Spence — APPROVED** [review]
Great work.

**Comment by h-m-f-t** [comment]
I agree with Matt. This has been a big chunk of bespoke, meaningful work that you've shepherded along so well, Erin. Excellent job!

**Comment by h-m-f-t** [comment]
I do have one outstanding q: do we know what `Last update of RDAP database` and `Reregistration` mean, exactly? I wonder if those are meaningful or if we should just strip them out.

**Comment by erinysong** [comment]
@h-m-f-t Those are events logged by Cloudflare. To my understanding `Last update of RDAP database` indicates when the RDAP database itself has been last updated (we can remove this in another ticket if it's not helpful). `Reregistration` I'm not as sure on but since it's consistently 1/3/2024, would that align with when we automatically renewed all domains or added them to the new registrar?

**Comment by erinysong** [comment]
Dealing with some issues with unverified commits in the middle of this branch. Will merge once I get those amended

**Comment by h-m-f-t** [comment]
Yes, that sounds about right. I could support removing both of those, but I think you're ready to be done here so it's ok if we do those later! =)

**Comment by erinysong** [comment]
@h-m-f-t we can remove those in this PR! I just remembered I set up an excluded fields map so we should be able to add those quickly to the excluded fields in events

**Pull request merged**
